src/internal/operators/share.ts

```diff
@@ -1,14 +1,51 @@
+/** @prettier */
 import { Observable } from '../Observable';
-import { multicast } from './multicast';
-import { refCount } from './refCount';
 import { Subject } from '../Subject';
 
-import { MonoTypeOperatorFunction } from '../types';
+import { MonoTypeOperatorFunction, OperatorFunction } from '../types';
+import { Subscription } from '../Subscription';
+import { from } from '../observable/from';
+import { lift } from '../util/lift';
+import { Subscriber } from '../Subscriber';
 
-function shareSubjectFactory() {
-  return new Subject<any>();
+interface ShareOptions<T, R> {
+  /**
+   * The factory used to create the subject that will connect the source observable to
+   * multicast consumers.
+   */
+  connector?: () => Subject<T>;
+  /**
+   * If true, the resulting observable will reset internal state on error from source and return to a "cold" state. This
+   * allows the resulting observable to be "retried" in the event of an error.
+   * If false, when an error comes from the source it will push the error into the connecting subject, and the subject
+   * will remain the connecting subject, meaning the resulting observable will not go "cold" again, and subsequent retries
+   * or resubscriptions will resubscribe to that same subject. In all cases, RxJS subjects will emit the same error again, however
+   * {@link ReplaySubject} will also push its buffered values before pushing the error.
+   */
+  resetOnError?: boolean;
+  /**
+   * If true, the resulting observable will reset internal state on completion from source and return to a "cold" state. This
+   * allows the resulting observable to be "repeated" after it is done.
+   * If false, when the source completes, it will push the completion through the connecting subject, and the subject
+   * will remain the connecting subject, meaning the resulting observable will not go "cold" again, and subsequent repeats
+   * or resubscriptions will resubscribe to that same subject.
+   */
+  resetOnComplete?: boolean;
+  /**
+   * If true, when the number of subscribers to the resulting observable reaches zero due to those subscribers unsubscribing, the
+   * internal state will be reset and the resulting observable will return to a "cold" state. This means that the next
+   * time the resulting observable is subscribed to, a new subject will be created and the source will be subscribed to
+   * again.
+   * If false, when the number of subscribers to the resulting observable reaches zero due to unsubscription, the subject
+   * will remain connected to the source, and new subscriptions to the result will be connected through that same subject.
+   */
+  resetOnUnsubscribe?: boolean;
```

Comment:
i was under the assumption that a complete/error will always result in an implicit `unsubscribe` afterwards. its not reflected in this description that an unsubscribe that follows a complete/error, will **not** result in a reset.

Reply:
> This means that the next time the resulting observable is subscribed to, a new subject will be created and the source will be subscribed to again.

This should also be added to the `resetOnError`/`resetOnComplete` descriptions. Maybe its worth to describe a `reset` independantly of these 3 configs in the share operator itself, so that the description of the reset params can be reduced to actual differences between them and therefore causing less mental friction.